### PR TITLE
add package mac ci

### PR DESCRIPTION
**Pull request by PikachuHy**

### CODE REVIEW

File: .github/workflows/package-macos.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Install Dependencies
+        run: |
+          brew install freetype libjpeg qt@5 cmake ccache curl
+          brew link qt@5
+      - name: Run Package Script
+        working-directory: ${{github.workspace}}
+        run: bash packages/macos/package.sh
+      - name: Compress
```
Comment (darcy-shen) [inline .github/workflows/package-macos.yml:25]: dmg is produced by `macdeployqt`

**Review by darcy-shen — CHANGES_REQUESTED** [review]

File: .github/workflows/package-macos.yml (hunk 1 above)
Comment (darcy-shen) [reply .github/workflows/package-macos.yml:25]: I guess that you should run `packages/macos/package.sh` locally step by step on macOS to understand what happened.

**Comment by PikachuHy** [comment]
```
cp /Applications/TeXmacs.app/Contents/Resources/share/TeXmacs/bin/gs Mogan.app/Contents/Resources/share/Xmacs/bin/
cp /Applications/Mogan.app/Contents/Resources/share/Xmacs/plugins/eukleides/bin/eukleides Mogan.app/Contents/Resources/share/Xmacs/plugins/eukleides/bin
```

why gs and eukleides not include on Mogan repo?

**Comment by darcy-shen** [comment]
For gs, see https://github.com/XmacsLabs/mogan/issues/294

For Eukleides, it is a plugin, we should maintain it outside Mogan Editor.

**Review by darcy-shen — APPROVED** [review]
LGTM

**Pull request merged**